eth-types/src/bytecode.rs

```diff
@@ -95,20 +95,20 @@ impl Bytecode {
     }
 
     /// Push
-    pub fn push(&mut self, n: usize, value: Word) -> &mut Self {
+    pub fn push(&mut self, n: u8, value: Word) -> &mut Self {
         debug_assert!((1..=32).contains(&n), "invalid push");
 
         // Write the op code
-        self.write_op_internal(OpcodeId::PUSH1.as_u8() + ((n - 1) as u8));
+        self.write_op((OpcodeId::push_n(n)).unwrap());
```

Comment:
I would recommend `expect` (along with a message) instead of `unwrap`. Both will panic if `n` is a bogus value, but we would have a better understanding of the panic.